scripts-available/CDB_FederatedServerListRemote.sql

```diff
@@ -0,0 +1,298 @@
+    -- having some privilege)."
+    -- https://www.postgresql.org/docs/11/infoschema-tables.html
+
+    -- Create local target schema if it does not exists
+    inf_schema name := 'information_schema';
+    remote_table name := 'tables';
+    local_schema name := @extschema@.__CDB_FS_Create_Schema(server_internal, inf_schema);
+BEGIN
+    -- Import the foreign `tables` if not done
+    IF NOT EXISTS (
+        SELECT * FROM pg_class
+        WHERE relnamespace = (SELECT oid FROM pg_namespace WHERE nspname = local_schema)
+        AND relname = remote_table
+    ) THEN
+        EXECUTE format('IMPORT FOREIGN SCHEMA %I LIMIT TO (%I) FROM SERVER %I INTO %I',
+                    inf_schema, remote_table, server_internal, local_schema);
+    END IF;
+
+    -- Note: in this context, schema names are not to be quoted
+    RETURN QUERY EXECUTE format($q$
+        SELECT table_name::name AS remote_table FROM %I.%I WHERE table_schema = '%s' ORDER BY table_name
+        $q$, local_schema, remote_table, remote_schema);
+END
+$func$
+LANGUAGE PLPGSQL VOLATILE PARALLEL UNSAFE;
+
+
+--
+-- List the columns in a remote PG schema
+--
+CREATE OR REPLACE FUNCTION @extschema@.__CDB_FS_List_Foreign_Columns_PG(server_internal name, remote_schema name)
+RETURNS TABLE(table_name name, column_name name, column_type text)
+AS $func$
+DECLARE
+    -- Import `columns` from the information schema
+    --
+    -- "The view columns contains information about all table columns (or view columns)
+    -- in the database. System columns (oid, etc.) are not included. Only those columns 
+    -- are shown that the current user has access to (by way of being the owner or having some privilege)."
+    -- https://www.postgresql.org/docs/11/infoschema-columns.html
+
+    -- Create local target schema if it does not exists
+    inf_schema name := 'information_schema';
+    remote_col_table name := 'columns';
+    local_schema name := @extschema@.__CDB_FS_Create_Schema(server_internal, inf_schema);
+BEGIN
+    -- Import the foreign `columns` if not done
+    IF NOT EXISTS (
+        SELECT * FROM pg_class
+        WHERE relnamespace = (SELECT oid FROM pg_namespace WHERE nspname = local_schema)
+        AND relname = remote_col_table
+    ) THEN
+        EXECUTE format('IMPORT FOREIGN SCHEMA %I LIMIT TO (%I) FROM SERVER %I INTO %I',
+                    inf_schema, remote_col_table, server_internal, local_schema);
+    END IF;
+
+    -- Note: in this context, schema names are not to be quoted
+    -- We join with the geometry columns to change the type `USER-DEFINED` 
+    -- by its appropiate geometry and srid
+    RETURN QUERY EXECUTE format($q$
+        SELECT 
+            a.table_name::name,
+            a.column_name::name,
+            COALESCE(b.column_type, a.data_type)::TEXT as column_type
+        FROM
+            %I.%I a
+        LEFT JOIN
+            @extschema@.__CDB_FS_List_Foreign_Geometry_Columns_PG('%s', '%s') b
```

Comment:
just curious, how are they reported in the information schema?

Reply:
As `USER-DEFINED`